examples/vision/swin_transformers.py

```diff
@@ -0,0 +1,557 @@
+print(f"x_test shape: {x_test.shape} - y_test shape: {y_test.shape}")
+
+plt.figure(figsize=(10, 10))
+for i in range(25):
+    plt.subplot(5, 5, i + 1)
+    plt.xticks([])
+    plt.yticks([])
+    plt.grid(False)
+    plt.imshow(x_train[i])
+plt.show()
+
+"""
+## Configure the hyperparameters
+
+In order to use each pixel as an individual input, you can set `patch_size` to (1, 1).
+We take inspirration from the original paper where they share the experimental settings
+for training on ImageNet-1K, keeping quite some of the settings same for this example.
+"""
+
+patch_size = (2, 2)  # 2-by-2 sized patches
+dropout_rate = 0.03  # Dropout rate
+num_heads = 8  # Attention heads
+embed_dim = 64  # Embedded dimensions
+num_mlp = 256  # MLP nodes
+qkv_bias = True  # Convert embedded patches to query, key, and values with a learnable additive value
+window_size = 2  # Size of attention window
+shift_size = 1  # Size of shifting
+image_dimension = 32
+
+num_patch_x = input_shape[0] // patch_size[0]
+num_patch_y = input_shape[1] // patch_size[1]
+
+learning_rate = 1e-3
+batch_size = 128
+num_epochs = 40
+validation_split = 0.1
+weight_decay = 0.0001
+label_smoothing = 0.1
+
+"""
+## Helper Functions
+
+We will now create two helper functions which can help us get a sequence of 
+patches from the image, allow us to merge patches to spatial frames and dropout.
+"""
+
+
+def window_partition(x, window_size):
+    _, height, width, channels = x.shape
+    patch_num_H = height // window_size
+    patch_num_W = width // window_size
+    x = tf.reshape(
+        x, shape=(-1, patch_num_H, window_size, patch_num_W, window_size, channels)
+    )
+    x = tf.transpose(x, (0, 1, 3, 2, 4, 5))
+    windows = tf.reshape(x, shape=(-1, window_size, window_size, channels))
+    return windows
+
+
+def window_reverse(windows, window_size, height, width, channels):
+    patch_num_H = height // window_size
+    patch_num_W = width // window_size
+    x = tf.reshape(
+        windows,
+        shape=(-1, patch_num_H, patch_num_W, window_size, window_size, channels),
+    )
+    x = tf.transpose(x, perm=(0, 1, 3, 2, 4, 5))
+    x = tf.reshape(x, shape=(-1, height, width, channels))
+    return x
+
+
+class DropPath(layers.Layer):
+    def __init__(self, drop_prob=None, **kwargs):
+        super(DropPath, self).__init__(**kwargs)
+        self.drop_prob = drop_prob
+
+    def call(self, x):
+        input_shape = tf.shape(x)
+        batch_size = input_shape[0]
+        rank = len(input_shape)
+        shape = (batch_size,) + (1,) * (rank - 1)
+        random_tensor = (1 - self.drop_prob) + tf.random.uniform(shape, dtype=x.dtype)
+        path_mask = tf.floor(random_tensor)
+        output = tf.math.divide(x, 1 - self.drop_prob) * path_mask
+        return output
+
+
+"""
+## MLP layer
+
+We will now create a simple multi-layered perceptron with 2 Dense and 2 Dropout layers.
+"""
+
+
+class Mlp(layers.Layer):
```

Comment:
This class is equivalent to 

```
keras.Sequential(
    [
            layers.Dense(filter_num[0]),
            layers.Activation(keras.activations.gelu),
            layers.Dropout(dropout_rate),
            layers.Dense(filter_num[1]),
            layers.Dropout(dropout_rate),
    ])
```

No need to wrap it.

Since it is only used once, I would recommend in-line the Sequential definition where the model is used.

Reply:
That makes sense but we get the values of `filter_num` only while inside the Swin Transformer block so I have defined the MLP inside the Swin Transformer class